app/src/main/java/io/appium/uiautomator2/handler/GetElementAttribute.java

```diff
@@ -79,70 +91,157 @@ public AppiumResponse safeHandle(IHttpRequest request) {
         } catch (UiAutomator2Exception e) {
             Logger.error(MessageFormat.format("Unable to retrieve attribute {0}", attributeName), e);
             return new AppiumResponse(getSessionId(request), WDStatus.UNKNOWN_ERROR, e);
-        } catch (ReflectiveOperationException | InvalidSelectorException e) {
+        } catch (ReflectiveOperationException e) {
             Logger.error("Can not access to method or field: ", e);
             return new AppiumResponse(getSessionId(request), WDStatus.UNKNOWN_ERROR, e);
         }
 
     }
 
-    private static class ContentSize {
-        int width;
-        int height;
-        int top;
-        int left;
-        int scrollableOffset;
-        int touchPadding;
+    private static int getScrollableOffset(AndroidElement uiScrollable) {
+        // get the bounds of the scrollable view
+        Rect bounds = getElementBoundsInScreen(uiScrollable);
 
-        ContentSize(Rect rect) {
-            width = rect.width();
-            height = rect.height();
-            top = rect.top;
-            left = rect.left;
+        // now scroll a bit back and forth in the view to populate the lastScrollData we need
+        int x1 = bounds.centerX();
+        int y1 = bounds.centerY() + MINI_SWIPE_PIXELS;
+        int x2 = x1;
+        int y2 = y1 - (MINI_SWIPE_PIXELS * 2);
+
+        // ensure that our xs and ys are within the bounds of the element
+        if (y1 > bounds.width()) {
+            y1 = bounds.width();
```

Comment:
It's better to [avoid edges of element](https://android.googlesource.com/platform/frameworks/testing/+/master/uiautomator/library/core-src/com/android/uiautomator/core/UiScrollable.java#635).

Reply:
yeah, good point. we have access to the touchPadding i think anyway.